src/sagemaker_containers/functions.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import inspect
+
+import six
+
+import sagemaker_containers as smc
+
+
+def matching_args(fn, dictionary):  # type: (function, collections.Mapping) -> dict
+    """Given a function fn and a dict dictionary, returns the function arguments that match the dict keys.
```

Comment:
I found this a bit confusing on first read-through. Maybe something like:

Given a function fn and a dict dictionary, returns a dictionary whose keys match (by name) the parameters of fn.

(I don't know if that's really better.)

Reply:
i like it better as well. Thanks